web/src/components/planner/Statistics.vue

```diff
@@ -0,0 +1,327 @@
+      </v-card>
+    </v-col>
+  </v-row>
+</template>
+
+  <script setup lang="ts">
+  import { computed, ref, watch } from 'vue'
+  import {
+    Factory,
+  } from '@/interfaces/planner/FactoryInterface'
+  import { formatNumber } from '@/utils/numberFormatter'
+  import {
+    getPartDisplayName,
+    hasMetricsForPart,
+  } from '@/utils/helpers'
+
+  const props = defineProps<{
+    factories: Factory[];
+    helpText: boolean;
+  }>()
+
+  // Initialize the 'hidden' ref based on the value in localStorage
+  const hidden = ref<boolean>(
+    localStorage.getItem('statisticsHidden') === 'true'
+  )
+
+  // Watch the 'hidden' ref and update localStorage whenever it changes
+  watch(hidden, newValue => {
+    localStorage.setItem('statisticsHidden', newValue.toString())
+  })
+
+  // Function to toggle visibility
+  const toggleVisibility = () => {
+    hidden.value = !hidden.value
+  }
+  // Initialize the 'hidden' ref based on the value in localStorage
+  const hiddenProducts = ref<boolean>(
+    localStorage.getItem('statisticsProductsHidden') === 'true'
+  )
+
+  // Watch the 'hidden' ref and update localStorage whenever it changes
+  watch(hiddenProducts, newValue => {
+    localStorage.setItem('statisticsProductsHidden', newValue.toString())
+  })
+
+  // Function to toggle visibility
+  const toggleProductsVisibility = () => {
+    hiddenProducts.value = !hiddenProducts.value
+  }
+
+  const getBuildingDisplayName = inject('getBuildingDisplayName') as (
+    part: string
+  ) => string
+
+  // This function calculates total number of buildings for each type of building and creates a new record to store that info in
+  const totalBuildingsByType = computed(() => {
+    const buildings: Record<
+      string,
+      {
+        name: string;
+        totalAmount: number;
+        powerPerBuilding: number;
+        totalPower: number;
+      }
+    > = {} // Explicitly define the type
+    props.factories.forEach(factory => {
+      Object.entries(factory.buildingRequirements).forEach(
+        ([key, requirement]) => {
+          if (!buildings[key]) {
+            // Initialize the building entry
+            buildings[key] = {
+              name: requirement.name,
+              totalAmount: 0,
+              powerPerBuilding: requirement.powerPerBuilding,
+              totalPower: 0,
+            }
+          }
+
+          // Accumulate the total amount and total power
+          buildings[key].totalAmount += requirement.amount
+          buildings[key].totalPower += requirement.totalPower
+        }
+      )
+    })
+
+    // return buildings
+    return Object.values(buildings).sort((a, b) =>
+      a.name.localeCompare(b.name)
+    )
+  })
+
+  // This function calculates total number of products produced
+  const allFactoryProducts = computed(() => {
+    const products: Record<
+      string,
+      { id: string, name: string; totalAmount: number; totalDifference: number }
+    > = {}
+
+    props.factories.forEach(factory => {
```

Comment:
Same with this, should be abstracted to it's own exported function and injected with the factories